docs/networking/simpleimpl.md

```diff
@@ -31,59 +27,35 @@ If your mod does not require the other side to have a specific network channel,
-----------------
-
-There are a couple things to highlight in a packet handler. A packet handler has both the message object and the network context available to it. The context allows access to the player that sent the packet (if on the server), and a way to enqueue thread-safe work.
-
-```java
-public static void handle(MyMessage msg, Supplier<NetworkEvent.Context> ctx) {
-  ctx.get().enqueueWork(() -> {
-    // Work that needs to be thread-safe (most work)
-    ServerPlayer sender = ctx.get().getSender(); // the client that sent this packet
-    // Do stuff
-  });
-  ctx.get().setPacketHandled(true);
-}
-```
-
-Packets sent from the server to the client should be handled in another class and wrapped via `DistExecutor#unsafeRunWhenOn`.
+Next, we must declare the types of messages that we would like to send and receive. This is a example:
 
 ```java
-// In Packet class
-public static void handle(MyClientMessage msg, Supplier<NetworkEvent.Context> ctx) {
-  ctx.get().enqueueWork(() ->
-    // Make sure it's only executed on the physical client
-    DistExecutor.unsafeRunWhenOn(Dist.CLIENT, () -> () -> ClientPacketHandlerClass.handlePacket(msg, ctx))
-  );
-  ctx.get().setPacketHandled(true);
-}
-
-// In ClientPacketHandlerClass
-public static void handlePacket(MyClientMessage msg, Supplier<NetworkEvent.Context> ctx) {
-  // Do stuff
-}
+private static final int PROTOCOL_VERSION = 1;
+public static final SimpleChannel INSTANCE = ChannelBuilder.named(new ResourceLocation("mymodid","main"))
+            .networkProtocolVersion(PROTOCOL_VERSION).
+            .acceptedVersions((status, version) -> PROTOCOL_VERSION==version)
+            .simpleChannel()
+            //register messages
+            .messageBuilder(MyMessage.class, NetworkDirection.PLAY_TO_SERVER)//The first parameter is the type of the message.The second of the parameter is the direction which will be asserted before any processing of this message occurs.
+            .decoder(MyMessage::new)//a `Function<FriendlyByteBuf, MSG>` responsible for decoding the message from the provided `FriendlyByteBuf`.
+            .encoder(MyMessage::encode)// a  `BiConsumer<MSG, FriendlyByteBuf>` responsible for encoding the message into the provided `FriendlyByteBuf`.
+            .consumerMainThread(MyMessage::handle)//Handle sent message on target side.a `BiConsumer<MSG, CustomPayloadEvent.Context>` responsible for handling the message itself on the main thread.
+            /*
+            consumerNetworkThread
+            The parameter of `consumerMainThread` is a `BiConsumer<MSG, CustomPayloadEvent.Context>` responsible for handling the 
+            message itself on the network thread.As of Minecraft 1.8 packets are by default handled on the network thread.
+            That means that your handler can _not_ interact with most game objects directly. Forge provides a convenient way to make your code execute on the main thread instead through `consumerMainThread`.
+            */
```

Comment:
Make this a sentence. Do not copy-paste from somewhere else since the context remains lost.